StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/FileHeaderTestBase.cs

```diff
@@ -0,0 +1,151 @@
+﻿namespace StyleCop.Analyzers.Test.DocumentationRules
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using TestHelper;
+    using Xunit;
+
+    /// <summary>
+    /// Base class for file header related unit tests.
+    /// </summary>
+    public abstract class FileHeaderTestBase : DiagnosticVerifier
+    {
+        /// <summary>
+        /// Gets the diagnostics that will be checked in the <see cref="TestEmptySourceAsync"/> test case.
+        /// </summary>
+        /// <value>
+        /// The diagnostics that should occur when the test case is executed.
+        /// </value>
+        protected virtual DiagnosticResult[] EmptySourceDiagnostics
```

Comment:
:exclamation: An empty source file should not produce diagnostics. This may deviate from the behavior of the reference StyleCop implementation, but it does not make sense to report diagnostics for an empty file.